brave/browser-laptop
# Windows - parse promo code from setup exe filename on first run

On first run after install, checks to see if provided a setup exe path via --squirrel-installer-path and if so, parses the promo code and writes it to a file in user-data.

Fix #12787 

**~Note: this is blocked whilst we wait for https://github.com/brave/muon-winstaller/pull/4 to be ~merged and~ published to npm and bumping the version in this PR's `package.json` to match, before merging~**

Submitter Checklist:


Test Plan:

This will involve building a Windows installer for Brave. Normally this requires some certificates for sigining. But this can be commented out for light testing, using this patch: https://gist.github.com/petemill/616b71b51e67e40848b77bc753f655ce and the simplified commands here:

 1. Clean `node_modules` and do fresh `npm install`, verifying `./node_modules/muon-winstaller` is fetched from github branch not npm
 2. Remove any build output `./Brave-XXX` or installer output `./dist` directories
 3. Uninstall Brave
 3. `CHANNEL=dev npm run build-package`
 4. `CHANNEL=dev npm run build-installer` <-- above patch required for this simplified command
 5. Rename installer at `./dist/Setup-Brave-x64.exe` to `Setup-Brave-x64-pem001.exe`
 6. Run the exe to install Brave

Observe:
- Brave installs correctly, and is run automatically after install
- A file 'promoCode' is created in the data directory, containing only the string `pem001`.

Reviewer Checklist:


Tests
